django/contrib/admin/filters.py

```diff
@@ -416,3 +416,47 @@ class RelatedOnlyFieldListFilter(RelatedFieldListFilter):
     def field_choices(self, field, request, model_admin):
         pk_qs = model_admin.get_queryset(request).distinct().values_list('%s__pk' % self.field_path, flat=True)
         return field.get_choices(include_blank=False, limit_choices_to={'pk__in': pk_qs})
+
+
+class BlankFieldListFilter(FieldListFilter):
+    def __init__(self, field, request, params, model, model_admin, field_path):
+        self.lookup_kwarg = '%s__blank' % field_path
+        self.lookup_val = params.get(self.lookup_kwarg)
+        super().__init__(field, request, params, model, model_admin, field_path)
+
+    def queryset(self, request, queryset):
+        if not self.used_parameters or self.lookup_kwarg not in self.used_parameters:
+            return queryset
+
+        del (self.used_parameters[self.lookup_kwarg])
+        blank = True if self.lookup_val == '1' else False
+
+        if not isinstance(self.field, models.CharField):
+            self.used_parameters['%s__isnull' % self.field_path] = blank
+            return super().queryset(request, queryset)
+
+        self.used_parameters['%s__isnull' % self.field_path] = True
+        # A blank CharField can be None or an empty string ('')
+        lookup_condition = models.Q(**self.used_parameters) | models.Q(**{self.field_path: ''})
```

Comment:
You can probably base all of these checks off `self.field.null` and `self.field.empty_strings_allowed` and avoid the `isinstance(isinstance(self.field, models.CharField))`.